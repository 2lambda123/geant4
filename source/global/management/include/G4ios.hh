// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: G4ios.hh,v 1.4 1999-11-23 15:00:04 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// ---------------------------------------------------------------
// GEANT 4 class header file
//
// G4ios.hh
//
// ---------------------------------------------------------------
#ifndef included_G4ios
#define included_G4ios

#if defined(OO_DDL_TRANSLATION)
/*
 * stdlib needs to be included before iostream.h
 * during oddlx runs to work around a parser
 * problem of AIX ooddlx v4.0.2 with some versions of
 * AIX system header files.
 */
#include <stdlib.h>
#endif

#include "g4std/iostream"
#include "G4Types.hh"

extern G4std::ostream G4cout;
extern G4std::ostream G4cerr;

#define G4cin G4std::cin
#define G4endl G4std::endl

#endif
