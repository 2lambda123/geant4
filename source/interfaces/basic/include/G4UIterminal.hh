//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UIterminal.hh,v 1.8 2003-06-16 16:55:57 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UIterminal_h
#define G4UIterminal_h 1

#include <fstream>
#include "G4UImanager.hh"
#include "G4VBasicShell.hh"
#include "G4VUIshell.hh"

//
//   Description:
//
//   This class inherits the class G4UIsession.
//   This is the class to use a character-terminal sesion.
//
//   Usage:  
//       G4UIsession* terminalSession = new G4UIterminal; 
//   or  G4UIsession* terminalSession = new G4UIterminal(new your-shell); 
//
//     A character-terminal session  "terminalSession" is instantiated.
//     G4cout stream is redirected by default to the constructed instance.
//
//   terminalSession-> SessionStart(); // "terminalSession" is started.
//   delete terminalSession;           // "terminalSession"  is deleted.
//
//
//   In default(no arguments are given), csh-like shell is instantiated.
//   If you want to use another shell (e.g. tcsh-like), you can give
//   your favorite shell in an argument of the constructor.
//
//   Which shell? / How to define your own shell?
//   Currently two kinds of shells,
//                   G4UIcsh / G4UItcsh
//   , are presented. 
//   They inherit the abstract base class, G4VUIshell.
//   In order to define your own shell, 
//     - Define your own shell class derived from G4VUIshell.
//     - Implement GetCommandLine() method (pure virtual).
//     - Add more functionality, if need.
//
//   For more detail, see source codes.
//

class G4UIterminal : public G4VBasicShell {
private:
  G4UImanager* UI;
  // shell
  G4VUIshell* shell;

  // program states
  G4bool iExit;
  G4bool iCont;

public:
  G4UIterminal(G4VUIshell* aShell=0);
  ~G4UIterminal();

  void SetPrompt(const G4String& prompt);

private:
  void ExecuteCommand(G4String aCommand);
  G4String GetCommand(const char* msg=0);
  G4bool GetHelpChoice(G4int& aInt);
  void ExitHelp();
  
public:
  // These methods are implementation of corresponding virtual methods
  // of G4UIsession class.
  virtual G4UIsession* SessionStart();  
  virtual void PauseSessionStart(G4String msg);
  virtual G4int ReceiveG4cout(G4String coutString);
  virtual G4int ReceiveG4cerr(G4String cerrString);

};

#endif

