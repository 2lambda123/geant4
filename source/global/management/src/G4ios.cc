//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
//
// $Id$
//
// 
// --------------------------------------------------------------
//      GEANT 4 class implementation file
//
// G4ios.cc
//
// History 1998 Nov. 3 Masayasu Nagamatu

#include "G4ios.hh"
#include "G4strstreambuf.hh"

G4ThreadLocal G4strstreambuf *G4coutbuf_G4MT_TLS_ = 0;
G4ThreadLocal G4strstreambuf *G4cerrbuf_G4MT_TLS_ = 0;
G4ThreadLocal std::ostream *G4cout_G4MT_TLS_ = 0;
G4ThreadLocal std::ostream *G4cerr_G4MT_TLS_ = 0;
#define G4coutbuf (*G4coutbuf_G4MT_TLS_)
#define G4cerrbuf (*G4cerrbuf_G4MT_TLS_)
#define G4cout (*G4cout_G4MT_TLS_)
#define G4cerr (*G4cerr_G4MT_TLS_)

void G4iosInitialization()
{
  if (G4coutbuf_G4MT_TLS_ == 0) G4coutbuf_G4MT_TLS_ = new G4strstreambuf;
  if (G4cerrbuf_G4MT_TLS_ == 0) G4cerrbuf_G4MT_TLS_ = new G4strstreambuf;
  if (G4cout_G4MT_TLS_ == 0) G4cout_G4MT_TLS_ = new std::ostream(G4coutbuf_G4MT_TLS_);
  if (G4cerr_G4MT_TLS_ == 0) G4cerr_G4MT_TLS_ = new std::ostream(G4cerrbuf_G4MT_TLS_);
}

void G4iosFinalization()
{
  delete G4cout_G4MT_TLS_;
  delete G4cerr_G4MT_TLS_; 
  delete G4coutbuf_G4MT_TLS_;
  delete G4cerrbuf_G4MT_TLS_;
}

void __attribute__ ((constructor)) my_init(void)
{
  G4iosInitialization();
  //printf("G4ios is initialized\n");
}

void __attribute__ ((destructor)) my_fini(void)
{
  G4iosFinalization();
  //printf("G4ios is finalized\n");
}
