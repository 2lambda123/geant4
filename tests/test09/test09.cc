// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: test09.cc,v 1.1 1999-01-08 16:35:23 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "Tst09DetectorConstruction.hh"
#include "Tst09RunAction.hh"
#include "Tst09PrimaryGeneratorAction.hh"
#include "Tst09PhysicsList.hh"
#include "Tst09SteppingAction.hh"
#include "Tst09TrackingAction.hh"

#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4RunManager.hh"

#include "G4ios.hh"

int main(int argc,char** argv) {

  // Set the default random engine to RanecuEngine
  RanecuEngine defaultEngine;
  HepRandom::setTheEngine(&defaultEngine);

  // Run manager
  G4RunManager * runManager = new G4RunManager;

  // UserInitialization classes
  runManager->SetUserInitialization(new Tst09DetectorConstruction);
  runManager->SetUserInitialization(new Tst09PhysicsList);

  // UserAction classes
  runManager->SetUserAction(new Tst09RunAction);
  runManager->SetUserAction(new Tst09PrimaryGeneratorAction);
  //runManager->SetUserAction(new Tst09SteppingAction);
  runManager->SetUserAction(new Tst09TrackingAction);

  if(argc==1)
  {
    // G4UIterminal is a (dumb) terminal.
    G4UIsession* session = new G4UIterminal;
    session->SessionStart();
    delete session;
  }
  else
  {
    G4UImanager* UImanager = G4UImanager::GetUIpointer();
    G4String command = "/control/execute ";
    G4String fileName = argv[1];
    UImanager->ApplyCommand(command+fileName);
  }

  delete runManager;
  return 0;
}

