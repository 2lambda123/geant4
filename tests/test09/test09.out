
        ############################################
        !!! WARNING - FPE detection is activated !!!
        ############################################

*************************************************************
 Geant4 version Name: geant4-09-03-ref-07    (25-June-2010)
                      Copyright : Geant4 Collaboration
                      Reference : NIM A 506 (2003), 250-303
                            WWW : http://cern.ch/geant4
*************************************************************

SimpleBox is now made of Pb
# /run/verbose 2
/run/initialize
Tst09PhysicsList::SetCuts:Default CutLength : 1 (mm)
/gun/direction 1 0 0
#
# gamma 500 MeV in Al
#
/mydet/SelectMaterial Al
SimpleBox is now made of Al
/gun/particle gamma
/gun/energy 500 MeV
/run/beamOn 100

phot:   for  gamma    SubType= 12
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       PhotoElectric :     Emin=          0 eV         Emax=   10 TeV

compt:   for  gamma    SubType= 13
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Klein-Nishina :     Emin=          0 eV         Emax=   10 TeV

conv:   for  gamma    SubType= 14
      Lambda tables from 1.022 MeV to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Bethe-Heitler :     Emin=          0 eV         Emax=   10 TeV

msc:   for e-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.04, stepLimitType: 1, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc93 :     Emin=          0 eV         Emax=   10 TeV

eIoni:   for  e-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :     Emin=          0 eV         Emax=   10 TeV

eBrem:   for  e-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :     Emin=          0 eV         Emax=   1 GeV
            eBremRel :     Emin=          1 GeV        Emax=   10 TeV

SynRad:  Incoherent Synchrotron Radiation
      good description for long magnets at all energies

eIoni:   for  e+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :     Emin=          0 eV         Emax=   10 TeV

eBrem:   for  e+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :     Emin=          0 eV         Emax=   1 GeV
            eBremRel :     Emin=          1 GeV        Emax=   10 TeV

annihil:   for  e+    SubType= 5
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            eplus2gg :     Emin=          0 eV         Emax=   10 TeV

msc:   for proton    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   2 MeV
          BetheBloch :     Emin=          2 MeV        Emax=   10 TeV

msc:   for GenericIon    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  anti_proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   2 MeV
          BetheBloch :     Emin=          2 MeV        Emax=   10 TeV

msc:   for kaon+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  kaon+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   1.05231 MeV
          BetheBloch :     Emin=    1.05231 MeV        Emax=   10 TeV

hIoni:   for  kaon-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   1.05231 MeV
          BetheBloch :     Emin=    1.05231 MeV        Emax=   10 TeV

msc:   for mu+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

muIoni:   for  mu+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   200 keV
          BetheBloch :     Emin=        200 keV        Emax=   1 GeV
        MuBetheBloch :     Emin=          1 GeV        Emax=   10 TeV

muBrems:   for  mu+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :     Emin=          0 eV         Emax=   10 TeV

muPairProd:   for  mu+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :     Emin=          0 eV         Emax=   10 TeV

muIoni:   for  mu-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   200 keV
          BetheBloch :     Emin=        200 keV        Emax=   1 GeV
        MuBetheBloch :     Emin=          1 GeV        Emax=   10 TeV

muBrems:   for  mu-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :     Emin=          0 eV         Emax=   10 TeV

muPairProd:   for  mu-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  pi+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   297.504 keV
          BetheBloch :     Emin=    297.504 keV        Emax=   10 TeV

msc:   for pi-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  pi-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   297.504 keV
          BetheBloch :     Emin=    297.504 keV        Emax=   10 TeV
#
# electron 500 MeV in Air
#
/mydet/SelectMaterial Air
SimpleBox is now made of Air
/gun/particle e-
/gun/energy 500 MeV
/run/beamOn 100

phot:   for  gamma    SubType= 12
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       PhotoElectric :     Emin=          0 eV         Emax=   10 TeV

compt:   for  gamma    SubType= 13
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Klein-Nishina :     Emin=          0 eV         Emax=   10 TeV

conv:   for  gamma    SubType= 14
      Lambda tables from 1.022 MeV to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
       Bethe-Heitler :     Emin=          0 eV         Emax=   10 TeV

msc:   for e-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.04, stepLimitType: 1, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc93 :     Emin=          0 eV         Emax=   10 TeV

eIoni:   for  e-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :     Emin=          0 eV         Emax=   10 TeV

eBrem:   for  e-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :     Emin=          0 eV         Emax=   1 GeV
            eBremRel :     Emin=          1 GeV        Emax=   10 TeV

SynRad:  Incoherent Synchrotron Radiation
      good description for long magnets at all energies

eIoni:   for  e+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
        MollerBhabha :     Emin=          0 eV         Emax=   10 TeV

eBrem:   for  e+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      LPM flag: 1 for E > 1 GeV
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               eBrem :     Emin=          0 eV         Emax=   1 GeV
            eBremRel :     Emin=          1 GeV        Emax=   10 TeV

annihil:   for  e+    SubType= 5
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            eplus2gg :     Emin=          0 eV         Emax=   10 TeV

msc:   for proton    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   2 MeV
          BetheBloch :     Emin=          2 MeV        Emax=   10 TeV

msc:   for GenericIon    SubType= 10
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 0
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  anti_proton    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   2 MeV
          BetheBloch :     Emin=          2 MeV        Emax=   10 TeV

msc:   for kaon+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  kaon+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   1.05231 MeV
          BetheBloch :     Emin=    1.05231 MeV        Emax=   10 TeV

hIoni:   for  kaon-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   1.05231 MeV
          BetheBloch :     Emin=    1.05231 MeV        Emax=   10 TeV

msc:   for mu+    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

muIoni:   for  mu+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   200 keV
          BetheBloch :     Emin=        200 keV        Emax=   1 GeV
        MuBetheBloch :     Emin=          1 GeV        Emax=   10 TeV

muBrems:   for  mu+    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :     Emin=          0 eV         Emax=   10 TeV

muPairProd:   for  mu+    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :     Emin=          0 eV         Emax=   10 TeV

muIoni:   for  mu-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   200 keV
          BetheBloch :     Emin=        200 keV        Emax=   1 GeV
        MuBetheBloch :     Emin=          1 GeV        Emax=   10 TeV

muBrems:   for  mu-    SubType= 3
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
              MuBrem :     Emin=          0 eV         Emax=   10 TeV

muPairProd:   for  mu-    SubType= 4
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          muPairProd :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  pi+    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
               Bragg :     Emin=          0 eV         Emax=   297.504 keV
          BetheBloch :     Emin=    297.504 keV        Emax=   10 TeV

msc:   for pi-    SubType= 10
      Lambda tables from 100 eV  to 10 TeV in 77 bins, spline: 1
      RangeFactor= 0.2, stepLimitType: 0, latDisplacement: 1
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
          UrbanMsc90 :     Emin=          0 eV         Emax=   10 TeV

hIoni:   for  pi-    SubType= 2
      dE/dx and range tables from 100 eV  to 10 TeV in 77 bins
      Lambda tables from threshold to 10 TeV in 77 bins, spline: 1
      finalRange(mm)= 1, dRoverRange= 0.2, integral: 1, fluct: 1, linLossLimit= 0.01
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
            ICRU73QO :     Emin=          0 eV         Emax=   297.504 keV
          BetheBloch :     Emin=    297.504 keV        Emax=   10 TeV
#

